src/mscorlib/shared/System/Text/Encoding.cs

```diff
@@ -1089,7 +1089,7 @@ public unsafe string GetString(ReadOnlySpan<byte> bytes)
         {
             fixed (byte* bytesPtr = &MemoryMarshal.GetReference(bytes))
             {
-                return GetString(bytesPtr, bytes.Length);
+                return String.CreateStringFromEncoding((bytes.Length != 0) ? bytesPtr : (byte*)1, bytes.Length, this);
```

Comment:
How did you decide on 2 vs 1?

Regardless, maybe put them into named constants since they're used in multiple places?

Reply:
2 for chars and 1 for bytes, so that the pointer has natural alignment. It probably does not matter much. 1 everywhere would work as good probably.

Named constants cannot be pointers. It would need to be properties.

Maybe we can have `MemoryMarshal.GetNonNullReference` that encapsulates this? I will give it a try to see how it looks.